### PR TITLE
Cleanup: strict grammar, xmldocs, embedded readme, nullable annotations

### PR DESCRIPTION
**Pull request by sungam3r**

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -22,14 +22,13 @@
     <EnableNETAnalyzers>true</EnableNETAnalyzers>
     <NoWarn>$(NoWarn);1591</NoWarn> <!--TODO: temporary solution, remove-->
     <Nullable>enable</Nullable>
+    <PackageReadmeFile Condition="'$(IsPackable)' == 'true'">README.md</PackageReadmeFile>
```
Comment (sungam3r) [inline src/Directory.Build.props:25]: +1KB : 238 -> 239

File: src/Directory.Build.targets [hunk 2]
```diff
@@ -3,7 +3,6 @@
   <PropertyGroup Condition="'$(IsTestProject)' == 'true'">
     <CoverletOutputFormat>opencover</CoverletOutputFormat>
     <CoverletOutput>$(MSBuildThisFileDirectory)../.coverage/$(AssemblyName)/</CoverletOutput>
-    <Include>[GraphQL-Parser]*</Include>
```
Comment (sungam3r) [inline src/Directory.Build.targets:6]: Seems needlessly.

File: src/GraphQLParser/AST/GraphQLFragmentDefinition.cs [hunk 3]
```diff
@@ -3,13 +3,21 @@ namespace GraphQLParser.AST
     /// <summary>
     /// AST node for <see cref="ASTNodeKind.FragmentDefinition"/>.
     /// </summary>
-    public class GraphQLFragmentDefinition : GraphQLInlineFragment, INamedNode
```
Comment (sungam3r) [inline src/GraphQLParser/AST/GraphQLFragmentDefinition.cs:6]: Bad design here.

File: src/GraphQLParser/ParserContext.Parse.cs [hunk 4]
```diff
@@ -464,7 +464,7 @@ private GraphQLEnumTypeExtension ParseEnumTypeExtension()
         }
 
         // http://spec.graphql.org/October2021/#EnumValue
-        private GraphQLValue ParseEnumValue()
+        private GraphQLValue ParseEnumValue() // TODO: deal with enum value as ValueNode vs NameNode
```
Comment (sungam3r) [inline src/GraphQLParser/ParserContext.Parse.cs:467]: Will be fixed in further PR.

File: src/GraphQLParser/AST/GraphQLOperationDefinition.cs [hunk 5]
```diff
@@ -16,14 +19,14 @@ public class GraphQLOperationDefinition : ASTNode, IHasDirectivesNode, INamedNod
         /// Note that name may be <see langword="null"/> for anonymous query.
         /// Therefore, the compiler shows CS8766 warning about nullability mismatch.
         /// </summary>
+#pragma warning disable CS8766 // Nullability of reference types in return type doesn't match implicitly implemented member (possibly because of nullability attributes).
         public GraphQLName? Name { get; set; }
```
Comment (Shane32) [inline src/GraphQLParser/AST/GraphQLOperationDefinition.cs:23]: So mismatched to INamedNode.Name

Ugh.  Oh well.

File: src/GraphQLParser/ParserContext.Parse.cs [hunk 6]
```diff
@@ -668,7 +668,9 @@ private GraphQLFragmentDefinition ParseFragmentDefinition()
             def.Comment = GetComment();
             ExpectKeyword("fragment");
             def.Name = ParseFragmentName();
-            def.TypeCondition = ParseTypeCondition(optional: false);
+#pragma warning disable CS8601 // Possible null reference assignment.
+            def.TypeCondition = ParseTypeCondition(optional: false); // never returns null
+#pragma warning restore CS8601
```
Comment (Shane32) [inline src/GraphQLParser/ParserContext.Parse.cs:673]:
```suggestion
            def.TypeCondition = ParseTypeCondition(optional: false)!; // never returns null
```

**Review by Shane32 — APPROVED** [review]

File: src/Directory.Build.targets (hunk 2 above)
Comment (sungam3r) [reply src/Directory.Build.targets]: 😕 I don't understand this. How it works?

File: src/GraphQLParser/ParserContext.Parse.cs (hunk 6 above)
Comment (sungam3r) [reply src/GraphQLParser/ParserContext.Parse.cs:673]: 🤦‍♂️

**Pull request merged**